src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs

```diff
@@ -1257,31 +1257,103 @@ public override bool IsEquivalentTo(Type? other)
 
         #region Attributes
 
+        internal CorElementType GetCorElementType()
+        {
+            TypeCache cache = Cache;
+            if ((cache.Cached & (int)TypeCacheEntries.CorElementType) != 0)
+                return cache.CorElementType;
+
+            var type = this;
+            cache.CorElementType = RuntimeTypeHandle.GetCorElementType (new QCallTypeHandle(ref type));
+            Interlocked.MemoryBarrier ();
+            UpdateCached(TypeCacheEntries.CorElementType);
+            return cache.CorElementType;
+        }
+
+        internal TypeAttributes GetAttributes()
+        {
+            TypeCache cache = Cache;
+            if ((cache.Cached & (int)TypeCacheEntries.TypeAttributes) != 0)
+                return cache.TypeAttributes;
```

Comment:
Is this going to work with hot-reload?

Reply:
If hot reload can change the information that is cached here, then we need a way to either disable this, or a way to invalidate the cache. There is already a cache with some entries which could be affected by this.

Reply:
There's an issue https://github.com/dotnet/runtime/issues/50978 to add a MetadataUpdateHandler for mono to clear reflection caches.  (we also have caching in native that we need to clear, and that's not done yet, too). /cc @fanyang-mono